analysis/process.cc

```diff
@@ -857,14 +861,59 @@ int main(int argc, char** argv)
                                          for (auto& i : lep_idxs){
                                              for (auto& j : lep_idxs){
                                                  if ( i == j ) continue;
-                                                 if ( std::abs(vvv.Common_lep_pdgid()[i]) == 11 and std::abs(vvv.Common_lep_pdgid()[j]) == 11 ) 
-                                                      passTrigger |= (vvv.Common_HLT_DoubleEl() and vvv.Common_lep_p4()[i].pt() > 25. and vvv.Common_lep_p4()[j].pt() > 15.);
-                                                 else if ( std::abs(vvv.Common_lep_pdgid()[i]) == 13 and std::abs(vvv.Common_lep_pdgid()[j]) == 11 )
-                                                      passTrigger |= (vvv.Common_HLT_MuEG() and vvv.Common_lep_p4()[i].pt() > 25. and vvv.Common_lep_p4()[j].pt() > 15.);
-                                                 else if ( std::abs(vvv.Common_lep_pdgid()[i]) == 11 and std::abs(vvv.Common_lep_pdgid()[j]) == 13 )
-                                                      passTrigger |= (vvv.Common_HLT_MuEG() and vvv.Common_lep_p4()[i].pt() > 25. and vvv.Common_lep_p4()[j].pt() > 10.);
-                                                 else if ( std::abs(vvv.Common_lep_pdgid()[i]) == 13 and std::abs(vvv.Common_lep_pdgid()[j]) == 13 )
-                                                      passTrigger |= (vvv.Common_HLT_DoubleMu() and vvv.Common_lep_p4()[i].pt() > 20. and vvv.Common_lep_p4()[j].pt() > 10.);
+
+                                                 bool trig_ee, trig_em, trig_me, trig_mm;
+                                                 float pti = vvv.Common_lep_p4()[i].pt();
+                                                 float ptj = vvv.Common_lep_p4()[j].pt();
+                                                 float pt0 = pti > ptj ? pti : ptj;
+                                                 float pt1 = pti > ptj ? ptj : pti;
+
+                                                 if (vvv.Common_year() == 2016)
+                                                 {
+                                                     trig_ee = vvv.Common_HLT_Ele23_Ele12_CaloIdL_TrackIdL_IsoVL_DZ();
```

Comment:
The previous check was not checking in detail which trigger bit fired. Only if "MuEG" fired. and hence there was a case where Ele 23 Mu8 fired but the the electron the event only had 24.9 something GeV that ended up passing, because it thought that Mu23 Ele12 fired. (